oidc-exchange.py

```diff
@@ -162,6 +176,28 @@ def _get(name: str) -> str:  # noqa: WPS430
     )
 
 
+def event_is_third_party_pr() -> bool:
+    # Non-`pull_request` events cannot be from third-party PRs.
+    if os.getenv("GITHUB_EVENT_NAME") != "pull_request":
+        return False
+
+    if event_path := os.getenv("GITHUB_EVENT_PATH"):
+        try:
+            event = json.loads(Path(event_path).read_bytes())
+        except json.JSONDecodeError:
+            debug("unexpected: GITHUB_EVENT_PATH does not contain valid JSON")
+            return False
+
+        try:
+            return event["pull_request"]["head"]["repo"]["fork"]
+        except KeyError:
+            return False
+
+    # No GITHUB_EVENT_PATH indicates a weird GitHub or runner bug.
+    debug("unexpected: no GITHUB_EVENT_PATH to check")
+    return False
```

Comment:
Just realized this entire block could be dedented with a simple guard check inversion..
```suggestion
    event_path = os.getenv("GITHUB_EVENT_PATH")
    if not event_path:
        # No GITHUB_EVENT_PATH indicates a weird GitHub or runner bug.
        debug("unexpected: no GITHUB_EVENT_PATH to check")
        return False

    try:
        event = json.loads(Path(event_path).read_bytes())
    except json.JSONDecodeError:
        debug("unexpected: GITHUB_EVENT_PATH does not contain valid JSON")
        return False

    try:
        return event["pull_request"]["head"]["repo"]["fork"]
    except KeyError:
        return False
```